libs/logging/src/logger.ts

```diff
@@ -58,4 +87,92 @@ export class Logger {
       console.log(logLine);
     }
   }
+
+  buildCDFLog(
+    electionDefinition: ElectionDefinition,
+    rawLogFileContents: string,
+    machineId: string,
+    codeVersion: string,
+    currentUser: LoggingUserRole
+  ): string {
+    if (!CLIENT_SIDE_LOG_SOURCES.includes(this.source)) {
+      void this.log(LogEventId.LogConversionToCdfComplete, currentUser, {
+        message: 'The current application is not able to export logs.',
+        result: 'Log file not converted to CDF format.',
+        disposition: 'failure',
+      });
+      throw new Error('Can only export CDF logs from a frontend app.');
+    }
+
+    const allEvents: CdfEvent[] = [];
+    const logs = rawLogFileContents.split('\n').filter((l) => l !== '');
+    // eslint-disable-next-line no-debugger
+    let idx = 0;
+    for (const log of logs) {
```

Comment:
```suggestion
    for (const [idx, log] of logs.entries()) {
```

Or, even better, stream the log lines.